tests/Doctrine/Tests/DBAL/Functional/Ticket/GH3423Test.php

```diff
@@ -0,0 +1,116 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Tests\DBAL\Functional\Ticket;
+
+use Doctrine\DBAL\Connection;
+use Doctrine\DBAL\DBALException;
+use Doctrine\Tests\DbalFunctionalTestCase;
+
+/**
+ * @see https://github.com/doctrine/dbal/issues/3423
+ */
+class GH3423Test extends DbalFunctionalTestCase
+{
+    /** @var bool */
+    private static $tableCreated = false;
+
+    protected function setUp()
+    {
+        parent::setUp();
+
+        if ($this->connection->getDatabasePlatform()->getName() !== 'postgresql') {
```

Comment:
Are all tests of this patch related to postgresql only? Which bits could be made DB-agnostic?

Reply:
Well as they all target the deferrable constraint then they are only usefull for databases which supports them. From a quick search it looks like Oracle could be targeted as well, if you support it. SQLite, MySQL and SQLServer don't support it them.

Reply:
OracleDB is supported here, but I was wondering about whether the test could've been made generic. Anyway, no need for it if none of the tests is fully generic: resolving here.